assets/js/base/components/dropdown-selector/index.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useRef } from '@wordpress/element';
+import classNames from 'classnames';
+import Downshift from 'downshift';
+
+/**
+ * Internal dependencies
+ */
+import DropdownSelectorInput from './input';
+import DropdownSelectorInputWrapper from './input-wrapper';
+import DropdownSelectorMenu from './menu';
+import DropdownSelectorSelectedChip from './selected-chip';
+import './style.scss';
+
+/**
+ * State reducer for the downshift component.
+ * See: https://github.com/downshift-js/downshift#statereducer
+ */
+const stateReducer = ( state, changes ) => {
+	switch ( changes.type ) {
+		case Downshift.stateChangeTypes.keyDownEnter:
+		case Downshift.stateChangeTypes.clickItem:
+			return {
+				...changes,
+				highlightedIndex: state.highlightedIndex,
+				isOpen: true,
+				inputValue: '',
+			};
+		case Downshift.stateChangeTypes.blurInput:
+		case Downshift.stateChangeTypes.mouseUp:
+			return {
+				...changes,
+				inputValue: state.inputValue,
+			};
+		default:
+			return changes;
+	}
+};
+
+/**
+ * Component used to show an input box with a dropdown with suggestions.
+ */
+const DropdownSelector = ( {
+	attributeLabel = '',
+	className,
+	checked = [],
+	inputLabel = '',
+	isDisabled = false,
+	isLoading = false,
+	onChange = () => {},
+	options = [],
+} ) => {
+	const inputRef = useRef( null );
```

Comment:
The same `inputRef` value is implemented on multiple inputs.  Was that intentional?

Reply:
Good catch! That was probably a copy-&-paste error because `<DropdownSelectorInputWrapper>` was not using the `inputRef` prop. Removed it in 7f77564fe8681ad172ba2a4851d72e4308d9591e.